### PR TITLE
23 port matlab push recovery code to c++

### PR DESCRIPTION
**Pull request by NhiNguyencmt8**

### CODE REVIEW

**Review by dtbpkmte — CHANGES_REQUESTED** [review]
Change float to double

File: ros2/src/push_recovery/src/test_control.cc [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#include "push_recovery/control.h"
+
+int main() {
+  std::cout << "This is meant to use for testing";
+  PushRecoveryControl Ibrahim;
+  Eigen::VectorXd v(3);
+  v << 1, 2, 3;
+  std::cout << "v2 =" << std::endl << v(2) << std::endl;
+  Ibrahim.r1_ft_force = {1,  2, 3};
```
Comment (dtbpkmte) [inline ros2/src/push_recovery/src/test_control.cc:9]: Spacing, public member variables

File: ros2/src/push_recovery/src/control.cc [hunk 2]
```diff
@@ -0,0 +1,678 @@
...
+  Eigen::MatrixXd J_X_COM(1, 3), J_X_COM_dot(1, 3);
+  J_X_COM << result_COM.row(0);
+  J_X_COM_dot << result_COM.row(1);
+  Eigen::MatrixXd Torque_SMC_Linear_plus_angular_compensation(3, 1);
+  Eigen::MatrixXd mat_smc(3, 1);
+  mat_smc << SMCController(cop,  J_X_COM, J_X_COM_dot);
+  Torque_SMC_Linear_plus_angular_compensation << mat_m * mat_smc  +  mat_n;
+
+  Eigen::MatrixXd q_double_dot(3, 1);
+  q_double_dot = SMCPOstureCorrection();
```
Comment (dtbpkmte) [inline ros2/src/push_recovery/src/control.cc:662]: O

**Review by dtbpkmte — CHANGES_REQUESTED** [review]
Move push_recovery to huron/control instead of ros2

File: ros2/src/push_recovery/src/test_control.cc (hunk 1 above)
Comment (NhiNguyencmt8) [reply ros2/src/push_recovery/src/test_control.cc]: Si

File: ros2/src/push_recovery/src/control.cc (hunk 2 above)
Comment (NhiNguyencmt8) [reply ros2/src/push_recovery/src/control.cc]: :O

**Comment by NhiNguyencmt8** [comment]
> Move push_recovery to huron/control instead of ros2

**Review by dtbpkmte — APPROVED** [review]

**Review by dtbpkmte — APPROVED** [review]

**Pull request merged**